sycl/tools/sycl-sanitize/main.cpp

```diff
@@ -0,0 +1,57 @@
+//==------------ main.cpp - SYCL Sanitizer Tool ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "launch.hpp"
+#include "llvm/Support/CommandLine.h"
+
+#include <iostream>
+#include <string>
+
+using namespace llvm;
+
+int main(int argc, char **argv, char *env[]) {
+  cl::opt<std::string> TargetExecutable(
+      cl::Positional, cl::desc("<target executable>"), cl::Required);
+  cl::list<std::string> Argv(cl::ConsumeAfter,
+                             cl::desc("<program arguments>..."));
+
+  cl::ParseCommandLineOptions(argc, argv);
+
+  std::vector<const char *> NewEnv;
+
+  {
+    size_t I = 0;
+    while (env[I] != nullptr)
+      NewEnv.push_back(env[I++]);
+  }
+
+  NewEnv.push_back("XPTI_FRAMEWORK_DISPATCHER=libxptifw.so");
+  NewEnv.push_back("XPTI_SUBSCRIBERS=libsycl_sanitizer_collector.so");
+  NewEnv.push_back("XPTI_TRACE_ENABLE=1");
+  NewEnv.push_back(nullptr);
+
+  std::vector<const char *> Args;
+
+  Args.push_back(TargetExecutable.c_str());
+
+  for (auto &Arg : Argv) {
+    Args.push_back(Arg.c_str());
+  }
+
+  Args.push_back(nullptr);
+
+  int Err = launch(TargetExecutable.c_str(), Args, NewEnv);
```

Comment:
Perhaps `launch` should have an STL friendly interface instead of having all these raw pointers?

Reply:
Does it look better now?

PS Perhaps STL should have a process launch function?